tests/utils-root.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright (c) 2023 PHYTEC Messtechnik GmbH
+ */
+
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <gio/gio.h>
+#include "helper.h"
+#include "pu-utils.h"
+#include "pu-error.h"
+
+static void
+test_is_drive(EmptyDeviceFixture *fixture,
+              G_GNUC_UNUSED gconstpointer user_data)
+{
+    g_assert_true(pu_is_drive(fixture->loop_dev));
+}
+
+static void
+test_wait_for_partitions(void)
+{
+    g_autoptr(GError) error = NULL;
+    g_assert_true(pu_wait_for_partitions(&error));
+    g_assert_no_error(error);
+}
+
+static void
+test_partition_set_partuuid(EmptyDeviceFixture *fixture,
+                            G_GNUC_UNUSED gconstpointer user_data)
+{
+    gint wait_status;
+    g_autofree gchar *output;
+    g_autofree gchar *cmd = g_strdup_printf("bash script/create_partition %s",
+                                            fixture->loop_dev);
+    g_assert_true(g_spawn_command_line_sync(cmd, NULL, NULL,
+                                            &wait_status, &fixture->error));
+    g_assert_no_error(fixture->error);
+    g_assert_true(g_spawn_check_wait_status(wait_status, &fixture->error));
+    g_assert_no_error(fixture->error);
+    g_assert_true(pu_partition_set_partuuid(fixture->loop_dev, 1,
+                                            "428957a5-839d-45ae-adaf-4108c98a087b",
+                                            &fixture->error));
+    g_assert_no_error(fixture->error);
+
+    cmd = g_strdup_printf("blkid -s PARTUUID -o value %sp1", fixture->loop_dev);
```

Comment:
Have you considered using `blkid_partition_get_uuid()` instead of calling `blkid`?